static/app/views/settings/featureFlags/changeTracking/newProviderForm.tsx

```diff
@@ -87,37 +92,95 @@ export default function NewProviderForm({
       });
     },
     onError: error => {
-      const message = t('Failed to add provider or secret.');
-      handleXhrErrorResponse(message, error);
-      addErrorMessage(message);
+      clearIndicators();
+      const responseJSON = error.responseJSON;
+
+      // Check if there are field-specific errors for 'secret' or 'provider'
+      const hasFieldSpecificErrors = responseJSON?.secret || responseJSON?.provider;
+
+      if (!hasFieldSpecificErrors) {
+        // General error - pass to parent component
+        const message =
+          responseJSON?.detail ||
+          (Array.isArray(responseJSON?.non_field_errors) &&
+            responseJSON.non_field_errors[0]) ||
+          (Array.isArray(responseJSON?.nonFieldErrors) &&
+            responseJSON.nonFieldErrors[0]) ||
+          t('Failed to add provider or secret.');
+        handleXhrErrorResponse(message, error);
+        onError(message);
+      }
+      // else Form component handles the error
     },
   });
 
+  // if secret exists, updating; else adding
+  const getButtonLabel = () => {
+    if (existingSecret) {
+      return t('Update Provider');
+    }
+    return t('Add Provider');
+  };
+
+  const handleGoBack = useCallback(() => {
+    navigate(
+      normalizeUrl(`/settings/${organization.slug}/feature-flags/change-tracking/`)
+    );
+  }, [organization.slug, navigate]);
```

Comment:
if you move this back up between `const [selectedProvider, setSelectedProvider] = useState('');` and `  const {mutate: submitSecret, isPending} = useMutation< ...`  it'll make the overall diff a bit smaller and easier to read.